lib/safe_active_record/load.rb

```diff
@@ -34,13 +72,21 @@ def self.apply_load_patch(safe_query_mgr)
           undef_method original_method
 
           define_method original_method do |*args|
-            if safe_query_mgr.activated? && safe_query_mgr.intercept_load?
+            if safe_query_mgr.activated? && safe_query_mgr.intercept_load? && !SafeActiveRecord.skip_symbols_diff(args, original_method.to_s)
               pre_symbols = Symbol.all_symbols
 
               result = method(:"safe_ar_original_#{original_method}").call(*args)
 
-              delta = Symbol.all_symbols - pre_symbols
-              safe_query_mgr.add_safe_queries delta
+              if result
+                post_symbols = Symbol.all_symbols
+                delta = if post_symbols[pre_symbols.size - 1] == pre_symbols[-1]
```

Comment:
Can you add a comment to explain the logic in this section?

Reply:
Done thanks!